utils/lib/grid_mapping.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python3
+import sqlite3
+import argparse
+import os
+import itertools
+
+# =============================================================================
+
+
+class GridLocMap(object):
+    """
+    This class preforms forward (physical -> VPR) and backward (VPR -> physical)
+    grid location mapping.
+    """
+
+    def __init__(self, fwd_loc_map, bwd_loc_map):
```

Comment:
Should this just take a connection to the database?

Reply:
There is a static method `load_from_database` which loads the grid map from database and returns an instance of the `GridLocMap` class.